cats/cats-core/src/main/java/com/netflix/spinnaker/cats/cache/CompositeCache.java

```diff
@@ -125,6 +127,25 @@ public CacheData get(String type, String id, CacheFilter cacheFilter) {
         return getAll(type, Arrays.asList(identifiers));
     }
 
+    @Override
+    public Map<String, Collection<CacheData>> getAllByApplication(String type, String application) {
+        return caches.iterator().next().getAllByApplication(type, application);
+    }
+
+    @Override
+    public Map<String, Collection<CacheData>> getAllByApplication(String type,
+                                                                  String application,
+                                                                  CacheFilter filter) {
+        return caches.iterator().next().getAllByApplication(type, application, filter);
+    }
+
+    @Override
+    public Map<String, Collection<CacheData>> getAllByApplication(Collection<String> types,
+                                                                  String application,
+                                                                  Map<String, CacheFilter> cacheFilters) {
+        return caches.iterator().next().getAllByApplication(types, application, cacheFilters);
+    }
```

Comment:
I think we should iterate all caches here, rather than blindly selecting the first, so that behavior of the class stays consistent across all of the methods. In the case of these specific methods being SQL-only, we'd probably do well to filter out any caches that don't successfully implement the new methods (via try/catch, likely).

Reply:
Made it consistent but left out the try/catch. The sql config only allows `CompositeCache` to be write up with a single provider of type `SqlProviderCache`. We don't actually want the functionality of `CompositeClass`, it just happens to get wired by default in all cases currently. Will address that as a separate issue.